### PR TITLE
Perf: cache iconv decoder

### PR DESCRIPTION
**Pull request by sandinmyjoints**
While getting an iconv decoder isn't particularly slow, I noticed it happens a lot when decoding lots of varchar rows. From internal testing, with 5000 rows of a table containing two varchar utf8mb3 columns (length 100 and 500), I'm seeing about 12% speedup with this change.

### CODE REVIEW

File: lib/parsers/string.js [hunk 1]
```diff
@@ -1,21 +1,42 @@
...
+  max: 500,
+});
+
+exports.decode = function (buffer, encoding, start, end, options) {
   if (Buffer.isEncoding(encoding)) {
     return buffer.toString(encoding, start, end);
   }
 
-  const decoder = Iconv.getDecoder(encoding, options || {});
+  // Optimize for common case: encoding="short_string", options=undefined.
```
Comment (sandinmyjoints) [inline lib/parsers/string.js:15]: I figure encoding is probably constant for most users. For my use case, it's always cesu8 (for utf8mb3).

**Comment by sandinmyjoints** [comment]
It looks like the CI failures are a few timeouts waiting for MySQL to come up. Unrelated to changes.

**Comment by wellwelwel** [comment]
> It looks like the CI failures are a few timeouts waiting for MySQL to come up. Unrelated to changes. 

Hi @sandinmyjoints, about the OSX tests, please ignore them for now 🙋🏻‍♂️

I run the "re-run" from time to time until the tests runs to its end.

**Comment by sidorares** [comment]
lgtm, thanks @sandinmyjoints !

**Comment by wellwelwel** [comment]
A suggestion would be to require LRU as `const LRU = require('lru-cache').default` to improve the compatibility with Next/Nuxt, etc.

What do you think, @sandinmyjoints? 🙋🏻‍♂️

---

For context:
- #2004

**Comment by sandinmyjoints** [comment]
@wellwelwel: updated!

**Pull request merged**